source/fab/tools/psyclone.py

```diff
@@ -0,0 +1,78 @@
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+
+"""This file contains the tool class for PSyclone.
+"""
+
+from pathlib import Path
+from typing import Callable, List, Optional, TYPE_CHECKING, Union
+
+from fab.tools.categories import Categories
+from fab.tools.tool import Tool
+
+if TYPE_CHECKING:
```

Comment:
I can't agree more, I've really tried to avoid this, but both classes needs this for typing, so I couldn't use the usual easy solution of just locally importing the symbol :( See #314 .